datasette/views/database.py

```diff
@@ -129,12 +131,46 @@ async def data(
             extra_args["custom_time_limit"] = int(params["_timelimit"])
         if _size:
             extra_args["page_size"] = _size
-        results = await self.ds.execute(
-            database, sql, params, truncate=True, **extra_args
-        )
-        columns = [r[0] for r in results.description]
 
         templates = ["query-{}.html".format(to_css_class(database)), "query.html"]
+
+        # Execute query - as write or as read
+        if write:
+            if request.method == "POST":
+                params = await request.post_vars()
+                write_ok = await self.ds.databases[database].execute_write(
+                    sql, params, block=True
+                )
+                return self.redirect(request, request.path + '?_success=Query+executed_successfully')
```

Comment:
Passing `?_success` like this isn't necessarily the best approach. Potential improvements include:
- Signing this message so it can't be tampered with (I could generate a signing secret on startup)
- Using a cookie with a temporary flash message in it instead
- Using HTML5 history API to remove the `?_success=` from the URL bar when the user lands on the page
If I add an option to redirect the user to another page after success I may need a mechanism to show a flash message on that page as well, in which case I'll need a general flash message solution that works for any page.

Reply:
I could add itsdangerous as a signing dependency, which delightfully has zero dependencies of its own: https://pypi.org/project/itsdangerous